src/fragments/lib/auth/js/manageusers.mdx

```diff
@@ -4,45 +4,150 @@ This section covers the client-side development experience for password and user
-  .then((user) => {
-    return Auth.changePassword(user, 'oldPassword', 'newPassword');
-  })
-  .then((data) => console.log(data))
-  .catch((err) => console.log(err));
+async function changePassword (oldPassword, newPassword) {
+  try {
+    const user = await Auth.currentAuthenticatedUser();
+    const data = await Auth.changePassword(user, oldPassword, newPassword);
+    console.log(data);
+  } catch(err) {
+    console.log(err);
+  }
+};
 ```
+</Block>
+</BlockSwitcher>
 
 ### Forgot password
 
+<BlockSwitcher>
+<Block name="TypeScript">
+
+```ts
+// Send confirmation code to user's email
+async function forgotPassword(username: string) {
+  try {
+    const data = await Auth.forgotPassword(username);
+    console.log(data);
+  } catch(err) {
+    console.log(err);
+  }
+};
+
+// Collect confirmation code and new password
+async function forgotPasswordSubmit(username: string, code: string, new_password: string) {
+  try {
+    const data = await Auth.forgotPasswordSubmit(username, code, new_password);
+    console.log(data);
+  } catch(err) {
+    console.log(err);
+  }
+};
+```
+</Block>
+<Block name="JavaScript">
+
 ```javascript
 import { Auth } from 'aws-amplify';
 
 // Send confirmation code to user's email
-Auth.forgotPassword(username)
-  .then((data) => console.log(data))
-  .catch((err) => console.log(err));
-
-// Collect confirmation code and new password, then
-Auth.forgotPasswordSubmit(username, code, new_password)
-  .then((data) => console.log(data))
-  .catch((err) => console.log(err));
+async function forgotPassword(username) {
+  try {
+    const data = await Auth.forgotPassword(username);
+    console.log(data);
+  } catch(err) {
+    console.log(err);
+  }
+} 
+
+// Collect confirmation code and new password
+async function forgotPasswordSubmit(username, code, new_password) {
+  try {
+    const data = await Auth.forgotPasswordSubmit(username, code, new_password);
+    console.log(data);
+  } catch(err) {
+    console.log(err);
+  }
+}
 ```
+</Block>
+</BlockSwitcher>
 
 ### Complete new password
 
 The user is asked to provide the new password and required attributes during the first sign-in attempt if a valid user directory is created in Amazon Cognito. During this scenario, the following method can be called to process the new password entered by the user.
 
+<BlockSwitcher>
+<Block name="TypeScript">
+
+```ts
+import { Auth } from 'aws-amplify';
+import type { CognitoUser } from '@aws-amplify/auth';
+
+type AmplifyCognitoUser = CognitoUser & {
+  challengeParam: {
+    requiredAttributes: string[];
+  }
+};
+
+async function completeNewPassword(username: string, password: string) {
+  try {
+    const user: AmplifyCognitoUser = await Auth.signIn(username, password);
+
+    if (user.challengeName === 'NEW_PASSWORD_REQUIRED') {
+      const { requiredAttributes } = user.challengeParam; // the array of required attributes, e.g ['email', 'phone_number']
+      const loggedInUser = await Auth.completeNewPassword(
```

Comment:
Can we put the `Auth.completeNewPassword()` call in a different scope than the `Auth.signIn()` call?  When we call `Auth.signIn()`, it's not always guaranteed that we'll get a response with `user.challengeName === 'NEW_PASSWORD_REQUIRED'`.